pkg/provider/kubernetes/credentials.go

```diff
@@ -411,18 +411,46 @@ func createOrUpdateKubevirtSecret(ctx context.Context, seedClient ctrlruntimecli
 
 func createVSphereSecret(ctx context.Context, seedClient ctrlruntimeclient.Client, cluster *kubermaticv1.Cluster) (bool, error) {
 	spec := cluster.Spec.Cloud.VSphere
+	// for existing clusters we need to fetch existing credential data to set infra user and password if empty
+	// this is caused by a bug fix related to credential separation
+	// see https://github.com/kubermatic/kubermatic/issues/13534 and https://github.com/kubermatic/kubermatic/issues/13995
+	// for further information
+	secretKeySelector := provider.SecretKeySelectorValueFuncFactory(ctx, seedClient)
 
+	if val, _ := secretKeySelector(spec.CredentialsReference, resources.VsphereInfraManagementUserUsername); val == "" {
+		if val, _ := secretKeySelector(spec.CredentialsReference, resources.VsphereUsername); val != "" {
+			spec.Username = val
+		}
+	}
+
+	if val, _ := secretKeySelector(spec.CredentialsReference, resources.VsphereInfraManagementUserPassword); val == "" {
+		if val, _ := secretKeySelector(spec.CredentialsReference, resources.VspherePassword); val != "" {
+			spec.Password = val
+		}
+	}
```

Comment:
Looking at the code itself, I don't believe we need to modify anything in the `createVSphereSecret` function.